### PR TITLE
Add a dropdown refill

### PR DESCRIPTION
**Pull request by creuter**
* The dropdown has appropriate ARIA labels and roles
* Only one dropdown is open at a time
* The dropdown closes on escape, when clicking the toggle, or when clicking outside the menu

dropdown

### CODE REVIEW

**Comment by whmii** [comment]
I'd be interested i having refills in ES6. I'm not the one writing it tho. 😅

File: source/refills/dropdown.html.erb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="dropdown">
+  <button data-dropdown="toggle" aria-haspopup="true" aria-expanded="false" aria-controls="example-dropdown">
+    Click to toggle example
+    <span aria-hidden="true"> ▾</span>
+  </button>
+  <ul class="dropdown-menu" id="example-dropdown" data-dropdown="menu" aria-hidden="true" aria-label="Example Dropdown Menu">
```
Comment (tysongach) [inline source/refills/dropdown.html.erb]: Are we wrapping at 80?

File: source/refills/dropdown.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<div class="dropdown">
+  <button data-dropdown="toggle" aria-haspopup="true" aria-expanded="false" aria-controls="example-dropdown">
```
Comment (tysongach) [inline source/refills/dropdown.html.erb]: Are we wrapping at 80?

**Comment by creuter** [comment]
Thanks!

1. I'd considered that, along with keyboard navigation within the dropdown, but think it might depend on the type of content in the dropdown. I didn't want to be too opinionated
2. Interesting thinking there, I can see the merits on both sides.

I'll merge as is, and revisit later.

**Pull request merged**